tripy/tripy/frontend/utils.py

```diff
@@ -191,14 +192,32 @@ def wrapper(*args, **kwargs):
             all_args = utils.merge_function_arguments(func, *args, **kwargs)
 
             if preprocess_args is not None:
+
+                # Since a Python function can have at most one variadic arg and merge_function_arguments
+                # would list all the entries consecutively, it suffices to find the name and start index
+                # for the variadic arg to handle variadic args when preprocessing.
+                def find_variadic_name_and_start_idx(all_args: Sequence) -> Optional[Tuple[str, int]]:
+                    encountered_names = {}
+                    for i, (name, _) in enumerate(all_args):
+                        if name in encountered_names:
+                            return name, encountered_names[name]
+                        if name not in encountered_names:
+                            encountered_names[name] = i
+
+                    return None
```

Comment:
We actually have this information already when we call `get_positional_arg_names` in `merge_function_arguments`. Could we just return it from there?

Reply:
I added it into there. Maybe a little odd to throw in an extra return value but it avoids an extra loop.